### PR TITLE
Added swift support to worker

### PR DESCRIPTION
**Pull request by andybest**
Hey, I believe that this should be all that is necessary to add swift support to the workers?

### CODE REVIEW

**Comment by lidavidm** [comment]
@harikmenon images updated with this

**Comment by andybest** [comment]
I can upload my WIP swift starter package bot. Would you prefer a zip, or a repo link?

**Comment by andybest** [comment]
@lidavidm Added the changes you requested

**Comment by andybest** [comment]
@lidavidm Here is a simple bot to test with https://www.dropbox.com/s/rarpdmto5g12ljl/swift-bot-test.zip?dl=0

**Comment by lidavidm** [comment]
Thanks! Also feel free to submit a PR to add the starter kit to our set of starter kits when you feel it's ready.

File: admin/setup_worker_image.sh [hunk 1]
```diff
@@ -85,6 +85,12 @@ sudo -iu bot_compilation sdk install groovy
 wget -O julia.tgz https://julialang-s3.julialang.org/bin/linux/x64/0.6/julia-0.6.1-linux-x86_64.tar.gz
 tar xvzf julia.tgz
 
+# Swift
+wget -O swift.tar.gz https://swift.org/builds/swift-4.0.2-release/ubuntu1610/swift-4.0.2-RELEASE/swift-4.0.2-RELEASE-ubuntu16.10.tar.gz
+sudo tar -C /usr/local -xzf swift.tar.gz
+echo 'export PATH="$PATH:/usr/local/swift-4.0.2-RELEASE-ubuntu16.10/usr/bin"' | sudo -iu bot_compilation tee -a /home/bot_compilation/.profile
+sudo chmod o+r /usr/local/swift-4.0.2-RELEASE-ubuntu16.10/usr/lib/swift/CoreFoundation/module.modulemap
```
Comment (lidavidm) [inline admin/setup_worker_image.sh]: Whoops, guess it needs to read all those files (should've figured). So this should be 
```
sudo chmod -R o+r /usr/local/swift-4.0.2-RELEASE-ubuntu16.10/usr/lib/swift/CoreFoundation/
```

Sorry for the trouble!

**Comment by lidavidm** [comment]
It also keeps complaining that `MyBot` wasn't created. I will rebuild the images and look into whatever permission issues are happening when I get a chance…

**Comment by andybest** [comment]
@lidavidm Updated :)

**Comment by andybest** [comment]
@lidavidm It's possible that I messed up the build commands. Basically you should just be able to cd into the bot directory, run 'swift build --configuration release', then the bot will be in ".build/release/MyBot"

**Comment by lidavidm** [comment]
Yup, and the file is there, the compiler script just refuses to see it.

File: apiserver/worker/compiler.py [hunk 2]
```diff
@@ -336,6 +336,9 @@ def compile(self, bot_dir, globs, errors, timelimit):
     "Scala": [
         ["scalac"],
     ],
+    "Swift": [
+        ["swift", "build", "--configuration", "release"]
+    ],
```
Comment (lidavidm) [inline apiserver/worker/compiler.py]: Add as well:

```
["chmod", "-R", "777", ".build"],
```

File: apiserver/worker/compiler.py [hunk 3]
```diff
@@ -528,6 +531,11 @@ def compile(self, bot_dir, globs, errors, timelimit):
         [],
         [(["*.ss"], ChmodCompiler("Scheme"))]
     ),
+    Language("Swift", ".build/release/" + BOT, "Package.swift", 
+        ".build/release/" + BOT,
+        [],
+        [([""], ErrorFilterCompiler(comp_args["Swift"][0], filter_stderr="warning:"))]
+    ),
```
Comment (lidavidm) [inline apiserver/worker/compiler.py]: Change the list of commands to:

```
        [
            ([""], ErrorFilterCompiler(comp_args["Swift"][0], filter_stderr="warning:")),
            ([""], ExternalCompiler(comp_args["Swift"][1])),
        ],
```

**Comment by lidavidm** [comment]
@harikmenon Images updated (again).

**Comment by andybest** [comment]
Updated with the new compile commands

**Comment by andybest** [comment]
I also added detection for Package.swift to the UploadBot template

**Pull request merged**